### ISSUE TITLE
Where are the examples?

### ISSUE THREAD

**Issue by smontanaro** [pybind/pybind11#577]
I'm sorry to bug you with such a simple question, but where are the examples? I saw a now-closed issue ([#288](https://github.com/pybind/pybind11/issues/288)) about renaming all the example*.cpp files. I've downloaded the 2.0.0rc1 zip file. Can't find actual examples anywhere. For example, this find/grep yields no grep hits at all:

`find /path/to/pybind11/source -type f | xargs egrep -l 'example-modules|example11|example-eval'`

**Comment by jagerman** [comment]
They are named "test" now, as they provide some examples, but primarily act as the test suite.

**Comment by smontanaro** [comment]
So I'm starting to try and use the tests as examples. At the risk of making myself instantly unpopular (but begging your indulgence, as I am a Python programmer, not a C++ programmer), I'm going to suggest that if test cases are also to be used as examples, where possible, their output should be as straightforward as possible. Where they are explicitly meant to be test cases and not teaching examples, they should probably be marked as such (e.g., "Don't do it this way! We are trying to break things!")

The very first test module I looked at was test_inheritance.py, where I found the following:

`
roger = Rabbit('Rabbit')
assert roger.name() + " is a " + roger.species() == "Rabbit is a parrot"
`

I would have expected "Roger is a rabbit", but maybe that's just me. Looking at test_inheritance.cpp, I can see very clearly that rabbits are, in fact, parrots. I've never owned a rabbit, but I see them in my yard all the time, and while they run quite fast, they certainly don't fly, nor do they squawk or curse like pirates.

I suspect there was a good reason for this bit of obfuscation that has to do with this being more of a test case than a tutorial example (increasing code coverage, perhaps). If so, then they fall down a bit as learning examples, if for no other reason than that some of them appear to cause the reader to do a brief double-take.

If there is a desire to provide clear tutorial examples, I'm happy to try and provide some as I learn how things work and my C++-fu improves. They say the best way to learn something is to try and teach it...
